torch/_inductor/kernel/flex/flex_decoding.py

```diff
@@ -70,16 +84,27 @@ def _use_flex_decoding(query, kv_indices, value, kernel_options, enable_gqa) ->
         sympy.And(sympy.Gt(ratio, 0), sympy.Eq(ratio & (ratio - 1), 0))
     )
 
-    return (
+    out = (
         not force_flex
         and not kernel_options.get("OUTPUT_MAX", False)
         and short_query_length
+        and can_use_single_block_m
         and static_batch
         and static_num_heads
         and non_zero_length
         and valid_block_mask_num_heads
         and pw_of_two
     )
+    log.debug(
```

Comment:
noob q does this do anything weird w/ guards? I imagine no right

Reply:
I don't think so. The flags printed are all bool rather than sympy expressions. Even if they are sympy expressions, printing them should not add guards.

Reply:
Yeah I mean in theory we should probably make this change as we we should forsure have a non zero number of full blocks just need to figure out the fast way to check if the page is full. That is fine can you make a code comment about this